packages/blueprints-integration/src/userEditing.ts

```diff
@@ -37,9 +45,45 @@ export interface UserEditingDefinitionForm {
 	currentValues: Record<string, any>
 }
 
+/**
+ * A form based operation where the user first selects the type
+ * of form they want to use (i.e. Camera form vs VT form)
+ */
+export interface UserEditingDefinitionSourceLayerForm {
+	type: UserEditingType.SOURCE_LAYER_FORM
+	/** Id of this operation */
+	id: string
+	/** Label to show to the user for this operation */
+	label: ITranslatableMessage
+	/** The json schemas describing the form to display */
+	schemas: Record<string, UserEditingSourceLayer>
+	/** Current values to populate the form with */
+	currentValues: {
+		type: SourceLayerType
```

Comment:
why are the schemas defined with arbitrary string keys, and a looser coupling to sourceLayerType than these values are? They should use the same root key.
I don't think coupling the values to sourceLayerType is a good idea, there is no guarantee that the blueprints will do that internally, they could have two different ways of generating parts which are both 'Camera' parts with completely different schema/value structures.